src/jit/assertionprop.cpp

```diff
@@ -2215,9 +2215,14 @@ GenTreePtr Compiler::optVNConstantPropOnTree(BasicBlock* block, GenTreePtr stmt,
 #ifdef _TARGET_64BIT_           
             if (vnStore->IsVNHandle(vnCns))
```

Comment:
I would also suggest to add an assert in gentree.cpp!gtFoldExprConst() that none of the constants being folded is a an Icon handle when under prejit mode.